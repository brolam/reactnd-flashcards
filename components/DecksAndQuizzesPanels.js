import React from 'react'
import { StyleSheet } from 'react-native'
import { View, FlatList } from 'react-native'
import PropTypes from 'prop-types'
import DeckCardList from './DeckCardList'

export default function DecksAndQuizzesPanels({ decks }) {
  return (
    <View style={styles.container} >
      <View style={styles.containerDeckList}>
        <DeckCardList decks={decks} />
      </View>
    </View>
  )
}

DecksAndQuizzesPanels.propTypes = {
  decks: PropTypes.array.isRequired,
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    flex: 1,
  },
  containerDeckList: {
    width: '40%'
  }
})