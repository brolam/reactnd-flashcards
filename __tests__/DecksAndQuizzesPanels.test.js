import React from 'react'
import DecksAndQuizzesPanels from '../components/DecksAndQuizzesPanels'

test('renders without crashing', () => {
  const wrapper = shallow(<DecksAndQuizzesPanels decks={deckDummies} />);
  expect(wrapper).toMatchSnapshot();
});

const deckDummies = [
  { id: 'one-item', title: 'One Deck', amountOfCards: 10 },
  { id: 'two-item', title: 'Two Deck', amountOfCards: 11 },
  { id: 'three-item',title: 'Three Deck', amountOfCards: 12 }
]