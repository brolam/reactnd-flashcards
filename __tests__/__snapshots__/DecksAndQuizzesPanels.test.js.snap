// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "width": "40%",
      }
    }
  >
    <DeckCardList
      decks={
        Array [
          Object {
            "amountOfCards": 10,
            "id": "one-item",
            "title": "One Deck",
          },
          Object {
            "amountOfCards": 11,
            "id": "two-item",
            "title": "Two Deck",
          },
          Object {
            "amountOfCards": 12,
            "id": "three-item",
            "title": "Three Deck",
          },
        ]
      }
    />
    <PanelQuizzes
      quizzes={
        Array [
          Object {
            "question": "",
          },
        ]
      }
    />
  </View>
</View>
`;
